jboss/container/wildfly/launch/datasources/added/launch/datasource-common.sh

```diff
@@ -684,41 +683,17 @@ function inject_datasource() {
   NON_XA_DATASOURCE=$(find_env "${prefix}_NONXA" false)
 
   url=$(find_env "${prefix}_URL")
+  driver=$(find_env "${prefix}_DRIVER" )
+  checker=$(find_env "${prefix}_CONNECTION_CHECKER" )
+  sorter=$(find_env "${prefix}_EXCEPTION_SORTER" )
+  url=$(find_env "${prefix}_URL" )
+  if [ -n "$checker" ] && [ -n "$sorter" ]; then
+    validate=true
+  else
+    validate="false"
+  fi
 
-  case "$db" in
-    "MYSQL")
-      map_properties "jdbc:mysql" "${prefix}_XA_CONNECTION_PROPERTY_ServerName" "${prefix}_XA_CONNECTION_PROPERTY_Port" "${prefix}_XA_CONNECTION_PROPERTY_DatabaseName"
```

Comment:
@luck3y @jfdenise @yersan We are now not adding these XA connection properties, which causes the bats tests to fail. I am not sure if this should still happen during this state, or if it should kick in during the s2i phase.

Reply:
If it should happen during this stage, it is a bit odd since we need to select Port for MySql vs PortNumber for Postgres

Reply:
@kabir AFIAR this is supposed to have XA_ENABLED (I think it is) set or disabled in the caller -- if its set then the associated properties should be required. Is this not what is there now?